### ISSUE TITLE
nicotine crash, 

### ISSUE THREAD

**Issue by rottweilerslip** [Nicotine-Plus/nicotine-plus#1040]
Nicotine+ version:  
3.0.3-202104022248~
Operating System/Distribution: 
ubuntu20.04.1

### Describe the bug
Type: <class 'AttributeError'>
Value: 'NoneType' object has no attribute 'shutdown_notification'
Traceback:   File "/usr/lib/python3/dist-packages/gi/_ossighelper.py", line 106, in signal_notify
    def signal_notify(source, condition):
  File "/usr/lib/python3/dist-packages/pynicotine/gtkgui/frame.py", line 2467, in on_quit
    self.save_state()
  File "/usr/lib/python3/dist-packages/pynicotine/gtkgui/frame.py", line 2495, in save_state
    self.np.pluginhandler.shutdown_notification()

### Steps to reproduce the bug
open app try to search and then it crashes

### Additional context
Screenshots, logs, stacktraces or relevant information.

**Comment by mathiascode** [comment]
Thanks! I think the important traceback (the first error popup) is missing in this case. Could you send the initial error message that appears, if any?

Also, do you have any plugins enabled? What are the contents of the "[searches]" section in your config file (located in ~/.config/nicoine/config or ~./nicotine/config, depending on how old your installation is).

**Comment by rottweilerslip** [comment]
Hiya ! 

Fortunately (and by pure chance) I sorted the problem out by changing the WiFi connection to my phone's WiFi and it started as if it was a new account (I had previously autoremoved nicotine) so I can't share with you the initial message anymore... 
I had uninstalled and installed nicotine+ before hand and the same thing happened. 
I'm back on the normal WiFi connection and it is still working. 
I'm very sorry, I'm not able to tell you exactly what plugins are installed. I know I had to get python3 for nicotine+...
In the config file here is what is under Searches : `[searches]
expand_searches = True
group_searches = 1
maxresults = 50
re_filter = False
history = ['country teasers empire', 'chrome something rhythmic', 'chrome subway']
enablefilters = False
filters_visible = False
defilter = ['', '', '', '', 0, '']
filtercc = []
filterin = []
filterout = []
filtersize = []
filterbr = []
filtertype = []
search_results = True
max_displayed_results = 1000
max_stored_results = 1500
min_search_chars = 3
remove_special_chars = True
`
Hope that helps... 
 
Thanks !

**Comment by kevinwithat** [comment]
I'm experiencing the same issue and can't seem to figure it out...

**Comment by kevinwithat** [comment]
Nicotine+ has encountered a critical error and needs to exit. Please copy the following error and include it in a bug report:

Type: <class 'TypeError'>
Value: 'NoneType' object is not subscriptable
Traceback:   File "/usr/lib/python3/dist-packages/pynicotine/gtkgui/frame.py", line 651, in on_network_event
    self.np.events[i.__class__](i)
  File "/usr/lib/python3/dist-packages/pynicotine/pynicotine.py", line 1066, in file_download
    self.transfers.file_download(msg)
  File "/usr/lib/python3/dist-packages/pynicotine/transfers.py", line 1323, in file_download
    self.download_finished(msg.file, i)
  File "/usr/lib/python3/dist-packages/pynicotine/transfers.py", line 1912, in download_finished
    self.eventprocessor.shares.add_file_to_shared(newname)
  File "/usr/lib/python3/dist-packages/pynicotine/shares.py", line 684, in add_file_to_shared
    if file not in (i[0] for i in shared[vdir]):
  File "/usr/lib/python3/dist-packages/pynicotine/shares.py", line 684, in <genexpr>
    if file not in (i[0] for i in shared[vdir]):

**Comment by mathiascode** [comment]
> EDIT: I'm having similar issues running the latest version of Nicotine+ (3.0.3)
> elementary OS 5.1.7 Hera
> Ubuntu 18.04.4

This should no longer happen in the unstable version. Could you confirm this?

**Comment by kevinwithat** [comment]
> > EDIT: I'm having similar issues running the latest version of Nicotine+ (3.0.3)
> > elementary OS 5.1.7 Hera
> > Ubuntu 18.04.4
> 
> This should no longer happen in the unstable version. Could you confirm this?

Running 3.0.5.dev1 now and receiving a similar error 

Nicotine+ has encountered a critical error and needs to exit. Please copy the following error and include it in a bug report:

Type: <class 'TypeError'>
Value: 'NoneType' object is not subscriptable
Traceback:   File "/usr/lib/python3/dist-packages/pynicotine/gtkgui/frame.py", line 582, in on_network_event
    self.np.events[i.__class__](i)
  File "/usr/lib/python3/dist-packages/pynicotine/pynicotine.py", line 1067, in file_download
    self.transfers.file_download(msg)
  File "/usr/lib/python3/dist-packages/pynicotine/transfers.py", line 1323, in file_download
    self.download_finished(msg.file, i)
  File "/usr/lib/python3/dist-packages/pynicotine/transfers.py", line 1916, in download_finished
    self.eventprocessor.shares.add_file_to_shared(newname)
  File "/usr/lib/python3/dist-packages/pynicotine/shares.py", line 684, in add_file_to_shared
    if file not in (i[0] for i in shared[vdir]):
  File "/usr/lib/python3/dist-packages/pynicotine/shares.py", line 684, in <genexpr>
    if file not in (i[0] for i in shared[vdir]):

**Comment by mathiascode** [comment]
Seems like the unstable PPA didn't contain the fixes yet. A new build should be available in a few minutes.

**Comment by flipMWD** [comment]
Don't even need to search, immediately after starting the program it freezes with these Critical Errors:

```
Type: <class 'KeyError'>
Value: 'country'
Traceback: File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/frame.py", line 651, in on_network_event
    self.np.events[i.__class__](i)
  File "/usr/lib/python3.9/site-packages/pynicotine/pynicotine.py", line 1238, in join_room
    self.chatrooms.join_room(msg)
  File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/chatrooms.py", line 222, in join_room
    tab = ChatRoom(self, msg.room, msg.users, meta)
  File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/chatroooms.py", line 577, in __init__
    config["columns"]["chat_room"][room]["country"]["visible"] = False
```

```
Type: <class 'KeyError'>
Value: 'nicotine'
Traceback:   File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/frame.py", line 651, in on_network_event
    self.np.events[i.__class__](i)
  File "/usr/lib/python3.9/site-packages/pynicotine/pynicotine.py", line 1262, in user_left_room
    self.chatrooms.user_left_room(msg)
  File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/chatrooms.py", line 392, in user_left_room
    self.joinedrooms[msg.room].user_left_room(msg.username)
```

```
Type: <class 'KeyError'>
Value: 'nicotine'
Traceback:   File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/frame.py", line 651, in on_network_event
    self.np.events[i.__class__](i)
  File "/usr/lib/python3.9/site-packages/pynicotine/pynicotine.py", line 1262, in user_left_room
    self.chatrooms.user_left_room(msg)
  File "/usr/lib/python3.9/site-packages/pynicotine/gtkgui/chatrooms.py", line 392, in user_left_room
    self.joinedrooms[msg.room].user_left_room(msg.username)
```

**Comment by mathiascode** [comment]
@flipMWD Fixed in unstable, will be fixed in 3.0.5. This error happened when joining a new room while country flags were hidden.
